floodlight/io/datasets.py

```diff
@@ -799,3 +801,278 @@ def _download_matches_info(self) -> None:
+            false None will be returned instead of the event data objects. This will
+            improve performance considerably if only event data is required.
+
+
+        Returns
+        -------
+        match_data: Tuple[Tuple[Events, Teamsheets, Pitch], Tuple[dict[XY], dict[Code],
+        dict[Code], dict[teamsheets], Pitch]
+            Returns a tuple of shape (event_data, position_data) as returned by the
+            ``floodlight.io.dfl.read_event_data_xml()`` and
+            ``floodlight.io.dfl.read_position_data_xml()`` functions for the requested
+            match. If any of the arguments ``events`` or ``positions`` are set to False,
+            Noneis returned instead of `event_data` or `position_data`, repectively.
+        """
+
+        if match_id in ["J03WMX", "J03WN1"]:
+            competition = "DFL-COM-000001"
+        else:
+            competition = "DFL-COM-000002"
+
+        file_name_infos = os.path.join(
+            self._data_dir,
+            f"DFL_02_01_matchinformation_{competition}_DFL-MAT-{match_id}.{self._IDSSE_FILE_EXT}",
+        )
+
+        file_name_events = os.path.join(
+            self._data_dir,
+            f"DFL_03_02_events_raw_{competition}_DFL-MAT-{match_id}.{self._IDSSE_FILE_EXT}",
+        )
+
+        file_name_positions = os.path.join(
+            self._data_dir,
+            f"DFL_04_03_positions_raw_observed_{competition}_DFL-MAT-{match_id}.{self._IDSSE_FILE_EXT}",
+        )
+
+        if not os.path.isfile(file_name_infos):
+            raise FileNotFoundError(
+                f"Could not load file, check class description for valid match "
+                f"and segment values ({file_name_infos})."
+            )
+
+        # parse event data
+
+        if events is True:
+            data_objects_events = read_event_data_xml(
+                file_name_events, file_name_infos, teamsheet_home, teamsheet_away
+            )
+        else:
+            data_objects_events = None
+
+        if positions is True:
+            data_objects_positions = read_position_data_xml(
+                file_name_positions, file_name_infos, teamsheet_home, teamsheet_away
+            )
+        else:
+            data_objects_events = None
+
+        # assemble
+        match_data = (data_objects_events, data_objects_positions)
+
+        return match_data
+
+    @staticmethod
+    def get_pitch() -> Pitch:
+        """Returns a Pitch object corresponding to the DFL-data."""
+        return Pitch.from_template("dfl", length=105, width=68)
+
+    def _download_and_write(self, file_name, host_url) -> None:
+        """Downloads an archive file into temporary storage and
+        extracts the content to the file system.
+        """
+        file = f"{self._data_dir}/{file_name}"
+        response = requests.get(host_url, allow_redirects=True)
+        with open(file, "w", encoding="utf-8") as xml_file:
+            xml_file.write(response.text)
+        xml_file.close()
```

Comment:
I know its also "wrong" in the EIGD parser, but private and staticmethods go above (I believe its protected - private - static - public)

Reply:
Interesting! Fixed